websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,279 @@
+const presence = new Presence({
+		clientId: "970743721404530798"
+	}),
+	browsingTimestamp = Math.floor(Date.now());
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, hostname } = window.location;
+	switch (hostname) {
+		case "sonicretro.org": {
+			presenceData.smallImageKey = "home_512";
+			presenceData.smallImageText = "sonicretro.org";
+			if (search) {
+				presenceData.largeImageKey = "search";
+				if (search === "?s=")
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for "${
+						document.title
+							.split("You searched for ")[1]
+							.split(" - Sonic Retro")[0]
+					}"`;
+				}
+			} else if (
+				(pathname === ("/" || "/#") && !search) ||
+				(pathname.includes("/page/") && !search)
+			) {
+				presenceData.largeImageKey = "home";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("/category/")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking through ${document.title
+					.split("-")[0]
+					.toLowerCase()}`;
+			} else if (
+				document.title.match(
+					/January|February|March|April|May|June|July|August|September|October|November|December/g
+				)
```

Comment:
you probably wanna match the URL instead, assuming you're talking about pages of this nature: "https://sonicretro.org/2022/05/02/". You can use something like `/\d{4}\/\d{2}\/\d{2}/.test(document.location.pathname)`

Reply:
> you probably wanna match the URL instead, assuming you're talking about pages of this nature: "https://sonicretro.org/2022/05/02/". You can use something like `/\d{4}\/\d{2}\/\d{2}/.test(document.location.pathname)`

But u can also sort posts by just month, would I have to make another case?

Reply:
you can make the last part of the regex optional